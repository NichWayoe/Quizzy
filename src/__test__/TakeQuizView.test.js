import TakeQuizView from '../Views/TakeQuizView.react';
import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import Quiz from '../Model/Quiz.react';

  describe('Test for the takeQuizComponent', () => {
    let quiz;
  
    beforeAll(() => {
      quiz = new Quiz("work and play", "Politics", "Nick", ["what", "Ama"], 2, {1:1, 2:1},{1:["WW","SS","SSS","ERR"], 2:["DAS","DSDS","DS","RE"]},60,"DDDS","YES THIS IS A QUIZ")
    });  
    test('this tests whether author is displayed', () => {
    render(<TakeQuizView quiz={[quiz]}/>)
    expect(screen.queryByText(/Nick/))
    });
    test('test for description shown', () => {
        render(<TakeQuizView quiz={[quiz]}/>)
        expect(screen.queryByText(/YES THIS IS A QUIZ/))
        });
    
    test('test for quiz title shown', () => {
            render(<TakeQuizView quiz={[quiz]}/>)
            expect(screen.queryByText(/work and play/))
            });
    test('test button for start quiz displayed', () => {
                render(<TakeQuizView quiz={[quiz]}/>)
                expect(screen.queryByText(/start/))
                });
    it('renders correctly when there are quiz given items', () => {
                    const tree = renderer.create(<TakeQuizView quiz={[quiz]}/>).toJSON();
                    expect(tree).toMatchSnapshot();
                  });

    it('renders correctly when there are no items', () => {
                    const tree = renderer.create(<TakeQuizView />).toJSON();
                    expect(tree).toMatchSnapshot();
                  });
  });