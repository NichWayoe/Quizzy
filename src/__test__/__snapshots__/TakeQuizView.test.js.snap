// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for the takeQuizComponent renders correctly when there are no items 1`] = `
<div>
  <form>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "gap": "20px",
          "justifyContent": "flex-end",
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        <p>
          NaN:NaN:60
        </p>
      </div>
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1apck9s-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        Submit
        <span
          className="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
    <fieldset
      className="MuiFormControl-root css-1vgzwi7-MuiFormControl-root"
    />
  </form>
</div>
`;

exports[`Test for the takeQuizComponent renders correctly when there are quiz given items 1`] = `
<div>
  <form>
    <div
      style={
        Object {
          "display": "flex",
          "flexDirection": "row",
          "gap": "20px",
          "justifyContent": "flex-end",
        }
      }
    >
      <div
        style={
          Object {
            "fontSize": 20,
          }
        }
      >
        <p>
          NaN:NaN:60
        </p>
      </div>
      <button
        className="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-1apck9s-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        tabIndex={0}
        type="submit"
      >
        Submit
      </button>
    </div>
    <fieldset
      className="MuiFormControl-root css-1vgzwi7-MuiFormControl-root"
    />
  </form>
</div>
`;
